content/3. Procedurele-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/While/1. While.md

```diff
@@ -0,0 +1,82 @@
+---
+title: 1. While
+taxonomie:
+  - ib-19.2.Controlestructuur-Herhaling-While.PI
+  - ib-19.3.Controlestructuur-Herhaling-While.PI
+---
+
+> Meer informatie over [[content/2. Ondersteunende-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/While/1. While|While]]
+
+## Stappen in de Systematische Probleem Aanpak
+### 1. Doel
+Bepaal waarom een **While**-loop nodig is. Bijvoorbeeld: Het programma moet een taak herhalen totdat een specifieke voorwaarde wordt voldaan.
+
+### 2. Subdoelen
+Splits het probleem in subdoelen, zoals:
+	• Het vaststellen van de herhalingsconditie.
+	• Het bepalen van de acties die moeten worden uitgevoerd zolang de conditie waar is.
+
+> [!info] Tip
+> Bedenk bij het ontwerpen van een _While_-loop altijd: “Wat moet er gebeuren voordat de loop stopt?” Zorg ervoor dat de conditie duidelijk definieert wanneer de herhaling moet eindigen en dat je variabelen binnen de loop bijwerkt om een oneindige lus te voorkomen.
+
+> [!TIP] Casus
+> **Gegeven is de volgende tekst:**
+> Een gebruiker probeert in te loggen. Het programma vraagt steeds om een gebruikersnaam en wachtwoord totdat beide correct zijn ingevoerd. Zodra de juiste combinatie is gegeven, krijgt de gebruiker toegang tot het systeem en stopt het vragen naar inloggegevens.
+> 
+> **Zet deze tekst om in een tabel:**
+> 
+> | Conditie                                    | Actie                   |
+> | ------------------------------------------- | ----------------------- |
+> | Gebruikersnaam of wachtwoord onjuist                         | Vraag opnieuw om beide |
+> | Gebruikersnaam en wachtwoord juist  | Verleen toegang en stop loop        |
+
+### 3. Fasen
+1. **Parameters/Variabelen Identificeren**
+	**Actie**: Bepaal de variabelen die nodig zijn voor de controle in de While-loop.
+	**Resultaat**: Gedefinieerde variabelen, zoals input en correctPassword.
```

Comment:
Opmerking: ergens ontbreekt volgens mij iets. Je zegt 'gedefinieerde' variabelen. Nu vermoed ik dat input en correctPassword beide Strings zijn, maar is die vereiste al bekend? Wanneer een nieuwkomer while( input != "bacon") invoert, kan deze verbaasd reageren wanneer de compiler meldt dat 'input' naar niks verwijst. 

Kan je het als iets verwoorden als 'Vooraf-gedefinieerde vars, zoals String input, die je daarom hier met input kan gebruiken'? Kan het mij voorstellen dat dit snel veel tekst wordt, echter...